rpc/handlers.go

```diff
@@ -126,6 +128,12 @@ func New(bcReader blockchain.Reader, syncReader sync.Reader, n utils.Network,
 	}
 }
 
+// WithFilterLimit sets the maximum number of blocks to scan in a single call for event filtering.
+func (h *Handler) WithFilterLimit(limit uint) *Handler {
```

Comment:
I am not sure about this, should we make `limit` an argument to `New`?

Reply:
I can't say for sure, but here's the reasoning as to why I used `With...`

- `limit`  sounds to me like an optional value rather than a core component needed for initialization.

- `New` already contains a significant number of values being passed in.

Reply:
Ok, let's follow @joshklop 's advice and make this disabled by default by setting the default value to `math.MaxUint`.
and remove `if h.filterLimit != 0` and `if cfg.RPCMaxBlockScan > 0` checks before calling `WithLimit` on `EventFilter` and `Handler`